source/ranges.tex

```diff
@@ -6124,11 +6124,19 @@
     constexpr auto begin() const requires @\placeholder{simple-view}@<V>
     { return @\exposid{iterator}@<true>(ranges::begin(@\exposid{base_}@)); }
 
-    constexpr auto end() requires (!@\placeholder{simple-view}@<V>)
-    { return ranges::end(@\exposid{base_}@); }
+    constexpr auto end()
+    { return @\exposid{sentinel}@<false>{ranges::end(@\exposid{base_}@)}; }
 
-    constexpr auto end() const requires @\placeholder{simple-view}@<V>
-    { return ranges::end(@\exposid{base_}@); }
+    constexpr auto end() requires common_range<V>
+    { return @\exposid{iterator}@<false>{ranges::end(@\exposid{base_}@)}; }
+
+    constexpr auto end() const
+      requires range<const V>
```

Comment:
There seems to be no particular reason not to put this *requires-clause* on the previous line, which would be more consistent with the style of the other declarations in this class. (Ditto line 6138.)